src/main/java/org/example/services/impl/PaymentServiceImpl.java

```diff
@@ -0,0 +1,53 @@
+package org.example.services.impl;
+
+import lombok.Getter;
+import org.example.dto.PaymentDetailsDTO;
+import org.example.models.PaymentMethodType;
+import org.example.services.IPayment;
+import org.example.services.PaymentService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.HashMap;
+import java.util.List;
+
+@Service
+public class PaymentServiceImpl implements PaymentService {
+    private final HashMap<PaymentMethodType, IPayment> paymentMethods;
+
+    @Getter
+    private IPayment paymentMethod;
```

Comment:
Getters and setter in a service class violates the OOP principles. take the payment method type in the parameters instead of setting them separately.